File: library/src/main/java/com/alamkanak/weekview/WeekViewContainer.java
package com.alamkanak.weekview;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.ViewConfiguration;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.widget.NestedScrollView;

public class WeekViewContainer extends NestedScrollView {

    private boolean scrollable = true;
    private final int mScaledTouchSlop;
    private float startX;
    private float startY;

    public WeekViewContainer(@NonNull Context context) {
        this(context, null);
    }

    public WeekViewContainer(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public WeekViewContainer(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        mScaledTouchSlop = ViewConfiguration.get(context).getScaledTouchSlop();
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        if (!scrollable) return false;
        switch (ev.getAction()) {
            case MotionEvent.ACTION_DOWN:
                startX = ev.getX();
                startY = ev.getY();
                break;
            case MotionEvent.ACTION_MOVE:
                float distanceX = Math.abs(ev.getX() - startX);
                float distanceY = Math.abs(ev.getY() - startY);
                if (distanceX > mScaledTouchSlop && distanceX * 2f > distanceY) {
                    // 判断为横向滑动
                    return false;
                }
                break;
        }
        return super.onInterceptTouchEvent(ev);
    }

    public void setScrollable(boolean enabled) {
        scrollable = enabled;
    }
}

File: library/src/main/java/com/alamkanak/weekview/WeekViewContainer.kt
package com.alamkanak.weekview

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.view.ViewConfiguration
import androidx.core.widget.NestedScrollView
import kotlin.math.abs

class WeekViewContainer @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : NestedScrollView(context, attrs, defStyleAttr) {

    private val scaledTouchSlop: Int
    private var scrollable = true
    private var startX = 0f
    private var startY = 0f

    init {
        scaledTouchSlop = ViewConfiguration.get(context).scaledTouchSlop
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        if (!scrollable) return false
        when (ev.action) {
            MotionEvent.ACTION_DOWN -> {
                startX = ev.x
                startY = ev.y
            }
            MotionEvent.ACTION_MOVE -> {
                val distanceX = abs(ev.x - startX)
                val distanceY = abs(ev.y - startY)
                if (distanceX > scaledTouchSlop && distanceX * 2f > distanceY) {
                    // 判断为横向滑动
                    return false
                }
            }
        }
        return super.onInterceptTouchEvent(ev)
    }

    fun setScrollable(enabled: Boolean) {
        scrollable = enabled
    }
}
